hammertime/rules/status.py

```diff
@@ -111,13 +112,28 @@ def _match(self, response, soft_404_response):
                 return hashlib.md5(response.raw).digest() == soft_404_response["raw_content_hash"]
             else:
                 try:
-                    resp_hash = Simhash(response.content, filter=self.match_filter, token_size=self.token_size)
+                    content = self._read_response(response)
+                    resp_hash = Simhash(content, filter=self.match_filter, token_size=self.token_size)
                     return resp_hash.distance(Simhash(soft_404_response["content_simhash"])) < self.distance_threshold
                 except UnicodeDecodeError:  # response content is not text, cannot match text.
                     return False
         else:
             return False
 
+    def _read_response(self, response):
+        if response.truncated:
```

Comment:
Perhaps this needs to be moved to the response interface somehow. Maybe as a partial_content that would only get the truncated content.

https://github.com/delvelabs/hammertime/blob/master/hammertime/engine/aiohttp.py#L134